test/emmy/examples/double_pendulum_test.cljc

```diff
@@ -157,12 +158,13 @@
                 "  const _60 = _18 * _20 * _23 * _56;\n"
                 "  const _62 = _60 + _59 + _58 + _32 + _34;\n"
                 "  return [1.0, [(- l_1 * y05 * _46 + l_2 * y04) / (_18 * l_2 * m_2 * _57 + _18 * l_2 * m_1), (- l_2 * m_2 * y04 * _46 + l_1 * m_1 * y05 + l_1 * m_2 * y05) / (l_1 * _20 * _23 * _57 + l_1 * _20 * m_1 * m_2)], [(- g * _19 * _20 * m_1 * _23 * _56 * _28 - g * _19 * _20 * _24 * _56 * _28 + 2.0 * g * _19 * _20 * _22 * m_2 * _55 * _28 + 4.0 * g * _19 * _20 * m_1 * _23 * _55 * _28 + 2.0 * g * _19 * _20 * _24 * _55 * _28 - g * _19 * _20 * Math.pow(m_1, 3.0) * _28 - 3.0 * g * _19 * _20 * _22 * m_2 * _28 - 3.0 * g * _19 * _20 * m_1 * _23 * _28 - g * _19 * _20 * _24 * _28 - l_1 * l_2 * m_2 * y04 * y05 * _55 * _50 + _18 * m_1 * _26 * _46 * _50 + _18 * m_2 * _26 * _46 * _50 + _20 * m_2 * _25 * _46 * _50 - l_1 * l_2 * m_1 * y04 * y05 * _50 - l_1 * l_2 * m_2 * y04 * y05 * _50) / _62, (- g * _18 * _21 * _24 * _56 * _29 - 2.0 * g * _18 * _21 * m_1 * _23 * _57 * _29 + 2.0 * g * _18 * _21 * _24 * _55 * _29 - g * _18 * _21 * _22 * m_2 * _29 - g * _18 * _21 * _24 * _29 + l_1 * l_2 * m_2 * y04 * y05 * _55 * _50 - _18 * m_1 * _26 * _46 * _50 - _18 * m_2 * _26 * _46 * _50 - _20 * m_2 * _25 * _46 * _50 + l_1 * l_2 * m_1 * y04 * y05 * _50 + l_1 * l_2 * m_2 * y04 * y05 * _50) / _62]];")]
-              (c/compile-state-fn*
+              (c/compile-state-fn
                #(e/Hamiltonian->state-derivative
```

Comment:
I like this idea: the whole params thing is a complication and allowing them to vary currently requires departing from the book style of integration. We need a new interface for this. My current thought is that your animation code would appeal to `compile-state-fn` and `stream-integrator` directly, because `evolve` and `state-advancer` do not give you a way to connect the primitive array that react is guarding to the integrand. So my current plan is to say that users of the SICM book interface cannot get the benefit of `generic-params`.

But you may have suggested a way out with the additional keyword parameter, thusly:

"If you say `generic-params?`, that implies compiling, because the only way to do that is to symbolically un-nest the state-derivative. But you must then supply a primitive array which the compiled integrand will close over  *OR* but then I will return to you an allocated primitive array, over which the integrand has closed, that you can mutate" (my preference would be the former, in the hope that that contract is easier to negotiate with react.